pkg/broker/ingress/ingress_handler.go

```diff
@@ -57,16 +74,16 @@ func (h *Handler) Start(ctx context.Context) error {
 	}
 }
 
-func (h *Handler) serveHTTP(ctx context.Context, event cloudevents.Event, resp *cloudevents.EventResponse) error {
+func (h *Handler) receive(ctx context.Context, event cloudevents.Event, resp *cloudevents.EventResponse) error {
 	// Setting the extension as a string as the CloudEvents sdk does not support non-string extensions.
-	event.SetExtension(broker.EventArrivalTime, time.Now().Format(time.RFC3339))
+	event.SetExtension(broker.EventArrivalTime, cloudevents.Timestamp{Time: time.Now()})
```

Comment:
can you make sure we are not breaking the way we are reading the extension on the filter pod?
https://github.com/knative/eventing/blob/b53d3fdb4b5e5ba736072a7a971f495ec09d16fa/pkg/broker/filter/filter_handler.go#L248

Reply:
yeah it works.